# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
# Commit: df07abc82381
# Full Hash: df07abc82381d2a3067b7de63867a0181f497519
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-26 09:13:23
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P2 PRemoteWorkerDebuggerMangaer binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebuggerManager binding.
#   
#   PRemoteWorkerDebuggerManager is the IPC that handles the registration/unregistration of remote debuggers.
# ==============================================================================

diff -r 5d39af040a61 -r df07abc82381 dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Mar 25 14:33:27 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Mar 25 14:33:27 2025 +0000
@@ -8,6 +8,26 @@
 
 namespace mozilla::dom {
 
+/* static */
+RefPtr<RemoteWorkerDebuggerManagerParent>
+RemoteWorkerDebuggerManagerParent::CreateForProcess(
+    mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerChild>* aChildEp) {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
+
+  mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerParent> parentEp;
+  nsresult rv =
+      PRemoteWorkerDebuggerManager::CreateEndpoints(&parentEp, aChildEp);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return nullptr;
+  }
+
+  RefPtr<RemoteWorkerDebuggerManagerParent> actor =
+      MakeRefPtr<RemoteWorkerDebuggerManagerParent>();
+  parentEp.Bind(actor);
+
+  return actor;
+}
+
 RemoteWorkerDebuggerManagerParent::RemoteWorkerDebuggerManagerParent() {
   MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
 }